lib/modules/manager/bazel-module/parser.ts

```diff
@@ -0,0 +1,61 @@
+import { lang, query as q } from 'good-enough-parser';
+import { regEx } from '../../../util/regex';
+import { Ctx } from './context';
+import type { ValueFragment } from './fragments';
+import * as starlark from './starlark';
+
+const booleanValuesRegex = regEx(`^${starlark.booleanStringValues.join('|')}$`);
+const supportedRules = ['bazel_dep'];
+const supportedRulesRegex = regEx(`^${supportedRules.join('|')}$`);
+
+/**
+ * Matches key-value pairs:
+ * - `name = "foobar"`
+ * - `dev_dependeny = True`
+ **/
+const kvParams = q
+  .sym<Ctx>((ctx, token) => {
+    return Ctx.as(ctx).startAttribute(token.value);
+  })
+  .op('=')
+  .alt(
+    q.str((ctx, token) => {
+      return Ctx.as(ctx).addString(token.value);
+    }),
+    q.sym<Ctx>(booleanValuesRegex, (ctx, token) => {
+      return Ctx.as(ctx).addBoolean(token.value);
+    })
+  );
+
+const moduleRules = q
+  .sym<Ctx>(supportedRulesRegex, (ctx, token) => {
+    return Ctx.as(ctx).startRule(token.value);
+  })
+  .join(
+    q.tree({
+      type: 'wrapped-tree',
+      maxDepth: 1,
+      search: kvParams,
+      postHandler: (ctx, tree) => {
+        return Ctx.as(ctx).endRule();
+      },
+    })
+  );
+
+const rule = q.alt<Ctx>(moduleRules);
+
+const query = q.tree<Ctx>({
+  type: 'root-tree',
+  maxDepth: 16,
+  search: rule,
+});
+
+const starlarkLang = lang.createLang('starlark');
+
+export function parse(input: string, packageFile?: string): ValueFragment[] {
+  const parsedResult = starlarkLang.query(input, query, new Ctx());
+  if (parsedResult) {
+    return parsedResult.results;
+  }
+  return [];
```

Comment:
we should log a debug message when file contents is invalid

Reply:
I added a test that feeds the parser invalid contents. The `query` API does not appear to throw. The signature does return `Ctx | null`. 

@zharinov Should I be collecting parsing errors in the `Ctx`? If so, could you point me to the correct hook/API?

Reply:
Based upon a Slack chat with @zharinov, the `query` API works like regex where invalid content will not cause an error to be thrown. As a result, there is nothing to log here.

Reply:
I did add some logging in `extractPackageFile` if no package dependencies are found in a package file.